packages/editor/src/browser/editor-command.ts

```diff
@@ -35,10 +38,54 @@ export namespace EditorCommands {
 @injectable()
 export class EditorCommandContribution implements CommandContribution {
 
+    constructor(@inject(FileSystem) protected readonly fileSystem: FileSystem,
+                @inject(PreferenceService) protected readonly preferences: PreferenceService) {}
+
     registerCommands(registry: CommandRegistry): void {
         registry.registerCommand(EditorCommands.SHOW_REFERENCES);
         registry.registerCommand(EditorCommands.CONFIG_INDENTATION);
         registry.registerCommand(EditorCommands.INDENT_USING_SPACES);
         registry.registerCommand(EditorCommands.INDENT_USING_TABS);
+
+        registry.registerCommand(CommonCommands.ENABLE_AUTO_SAVE, {
+            isEnabled: () => !this.isAutoSaveOn(),
+            execute: () => this.switchAutoSave()
+        });
+        registry.registerCommand(CommonCommands.DISABLE_AUTO_SAVE, {
+            isEnabled: () => this.isAutoSaveOn(),
+            execute: () => this.switchAutoSave()
+        });
+    }
+
+    private async switchAutoSave(): Promise<void> {
+        const home = await this.fileSystem.getCurrentUserHome();
+        const folderPath = home.uri + "/.theia";
+        const settingsPath = folderPath + "/settings.json";
+        if (!await this.fileSystem.exists(folderPath)) {
+            this.fileSystem.createFolder(folderPath);
+        }
+        if (!await this.fileSystem.exists(settingsPath)) {
+            this.fileSystem.createFile(settingsPath);
+        }
+        const fileStat = await this.fileSystem.getFileStat(settingsPath);
+        const result = await this.fileSystem.resolveContent(settingsPath);
+        const content = result.content;
+        const preference: string = '"editor.autoSave" : ' + (this.isAutoSaveOn() ? '"off"' : '"on"');
+        if (content.length === 0) {
+            this.fileSystem.updateContent(fileStat, [{text: '{\n   ' + preference + '\n}'}]);
+            return;
+        }
+        let replace: string;
+        const pattern = new RegExp('"editor.autoSave"\\s?:\\s?("on"|"off")');
```

Comment:
jsonc-parser is already used to parse settings. it has utils to edit documents, too.